### PR TITLE
Invoicing Page Analytic Graphs and Detailed Expansion

### PR DESCRIPTION
**Pull request by jjessieshang**
Additional features so the user is able to gain insight into the meaning of the SOW, Service, and Project counts. Each count is an expandable component that helps the user visualize the numbers. Additionally, a click counting mechanism is associated with each component to track their usefulness.

https://user-images.githubusercontent.com/105069660/230691286-40fe83b6-6e2a-4f8b-a4b9-9a8ab8dc46e8.mov

Click analytics for /invoice page and each of the three clickable popups. When a popup component is clicked, the respective click count in database table `click_analytics` will be incremented by 1. When the invoice page is rendered, the respective click count will increment if the last rendered time is more than 1 minute ago
Screenshot 2023-04-08 at 2 15 01 PM

### CODE REVIEW

File: frontend/src/constants.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ export const appColor = {
   darkGray: "#909090",
 };
 
-export const backend = process.env.REACT_APP_BACKEND;
-// export const backend = "http://localhost:8080/"
+// export const backend = process.env.REACT_APP_BACKEND;
+export const backend = "http://localhost:8080/";
```
Comment (Harin329) [inline frontend/src/constants.js:15]: I think you forgot to uncomment this one :)

File: frontend/src/screens/invoice/invoice.js [hunk 2]
```diff
@@ -90,7 +106,31 @@ function Invoice() {
...
+  
+    if (!storedTime || currentTime - storedTime > 60000) {
+      {console.log("update")}
+      dispatch({
+        type: UPDATE_CLICKS,
+        payload: {
+          component_name: "billing_page"
+        },
+      });
+      localStorage.setItem("lastMountTime", JSON.stringify(currentTime));
```
Comment (Harin329) [inline frontend/src/screens/invoice/invoice.js:123]: Nice! 👍

File: frontend/src/screens/invoice/invoice.js [hunk 3]
```diff
@@ -90,7 +106,31 @@ function Invoice() {
...
   }, [dispatch, usersData, organizationData, projectData, costCenterData]);
 
+  useEffect(() => {
+    const storedTime = JSON.parse(localStorage.getItem("lastMountTime"));
+    const currentTime = new Date().getTime();
+    {console.log(storedTime)}
+    {console.log(currentTime)}
+  
+    if (!storedTime || currentTime - storedTime > 60000) {
+      {console.log("update")}
```
Comment (Harin329) [inline frontend/src/screens/invoice/invoice.js:116]: can probably remove this log

File: frontend/src/components/ProjectsAnalytics/index.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const uniqueProjects = Object.entries(projectsCostsSum)
+    .map(([fk_project_id, cost]) => ({ project: fk_project_id, cost }))
+    .sort((a, b) => b.cost - a.cost);
+  const categories = uniqueProjects.map(project => project.project);
+  const costData = uniqueProjects.map(project => project.cost);
+
+  var projectsChart = {
+      series: costData,
+      options: {
+        colors: [ "#5976E1", "#4E3340", "#CF6973", "#D4D145", "#ECCC4C",],
```
Comment (Harin329) [inline frontend/src/components/ProjectsAnalytics/index.js:32]: Let's put this into the appColor constant file, so we can easily match styling if we need to

File: frontend/src/components/SowAnalytics/index.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  const categories = Object.keys(projectsCount);
+  const data = Object.values(projectsCount);
+  var sowChart = 
+    { series: data,
+        options: {
+          chart: {
+            type: 'polarArea',
+          },
+          labels: categories,
+          colors: [ "#5976E1", "#4E3340", "#CF6973", "#D4D145","#ECCC4C",],
```
Comment (Harin329) [inline frontend/src/components/SowAnalytics/index.js:30]: appColor constant here as well

File: database/procedures/clickAnalytics.sql [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+DROP PROCEDURE IF EXISTS `createClickAnalytics`;
+DROP PROCEDURE IF EXISTS `update_count`;
+DROP PROCEDURE IF EXISTS `initialize_count`;
+
+DELIMITER $$
+
+CREATE PROCEDURE `createClickAnalytics` ()
+BEGIN
+
+CREATE TABLE `click_analytics` (
```
Comment (Harin329) [inline database/procedures/clickAnalytics.sql:12]: Let's put the create table call inside the init folder, so it's separate from the stored procedures!

File: database/procedures/clickAnalytics.sql [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+DELIMITER ;
+
+USE `labby`;
+
+CALL initialize_count('billing_page');
+CALL initialize_count('sow_chart');
+CALL initialize_count('services_chart');
+CALL initialize_count('projects_chart');
+SELECT*from click_analytics
```
Comment (Harin329) [inline database/procedures/clickAnalytics.sql:43]: was this put here intentionally?

**Review by Harin329 — CHANGES_REQUESTED** [review]
Overall it looks really good! Nice job with the graphs and also the localstorage. Just a few small styling and cleanup requests here

File: database/procedures/clickAnalytics.sql (hunk 7 above)
Comment (jjessieshang) [reply database/procedures/clickAnalytics.sql]: Thanks for catching these things Harin. Think I've made all the changes now.

**Review by Harin329 — APPROVED** [review]
Some merge conflicts that need to be resolved but code looks good.

**Pull request merged**